app/index.js

```diff
@@ -7,58 +7,50 @@ var PatternGenerator = yeoman.generators.Base.extend({
     'use strict';
     yeoman.generators.Base.apply(this, arguments);
 
+    this.argument('app_name', { type: String, required: true });
+    this.appname = this.app_name || this.appname;
     this.option('appPath', {
       desc: 'Name of application directory',
       type: 'String',
-      defaults: './',
+      defaults: this.appname,
       banner: 'some banner'
     });
-    this.argument('app_name', { type: String, required: false });
-    this.appname = this.app_name || this.appname;
-    this.env.options.appPath = this.options.appPath || './';
+    this.env.options.appPath = this.options.appPath;
     this.config.set('appPath', this.env.options.appPath);
-    this.indexFile = this.readFileAsString(path.join(this.sourceRoot(), 'index.html'));
   },
 
   writing: {
-    setupEnv: function () {
+
+    setup_base: function () {
       this.mkdir(this.env.options.appPath);
```

Comment:
Patternslib style guide? I prefer pep8 style more, because of better visual seperation.
However, will fix.